### ISSUE TITLE
Modern theme and more options

### ISSUE THREAD

**Issue by DeeliN221** [certik/theoretical-physics#105]
Just a suggestion, ReadTheDocs theme and Pydata theme both look nice and have many advanced functions, such as dark mode, sticky TOC,  and easily navigation. They are being used by many popular scientific projects like numpy, scipy, pandas, ...

If you guys like, I will create a PR then, with a "dark-mode" toggle button, sticky TOC, wider page, and quickly copy code-block function enabled.

**Comment by certik** [comment]
@m3lab-zzl that would be awesome, thank you. Go ahead!

**Comment by DeeliN221** [comment]
Right, I am working on this (the toolchain got me some trouble due to bad internet connection). The PR will come soon.

**Comment by DeeliN221** [comment]
## Preview the mathjax version (equations are rendered by mathjax on the fly)

- Current one:

![image](https://github.com/certik/theoretical-physics/assets/102862066/96b8a8c8-1113-4c50-ac68-1bf3d93fd3cc)

- Light ReadTheDocs theme:

![image](https://github.com/certik/theoretical-physics/assets/102862066/ae968a2a-b449-4bad-a460-3c76b935fb95)

- Dark ReadTheDocs theme:

![image](https://github.com/certik/theoretical-physics/assets/102862066/c0a4de5d-5691-40c4-827a-89d6eb46ddc8)

- Light Pydata theme:

![image](https://github.com/certik/theoretical-physics/assets/102862066/898a8a00-f656-4678-87e3-b20e811b8287)

- Dark Pydata theme:

![image](https://github.com/certik/theoretical-physics/assets/102862066/f0fb1317-eee6-405f-b0b3-a46437f28098)

**The new 4 options all have a user-friendly sidebar that makes the navigation easier, and a search box for quickly jumping to any interested part. The page width is no longer fixed at some weird width that would leave the right part blank. One final change is the code-block copy button and autowrap:**

![image](https://github.com/certik/theoretical-physics/assets/102862066/54ab5daa-998d-4c81-b53d-88ca57a6aaeb)

Pydata theme has a `Back to top` button, if you like this theme, we may need to change the project structure, as the top navigation bar can only display short titles.

- About the equations

As you mentioned before, Mathjax rendered equations look nicer (also visible in dark mode), and it took much less time to build, but the source code needs some modification to correctly generate the desired output. If you insist the dvisvg solution, equations will be invisible in dark mode.

See PR #106 for code change

**Comment by DeeliN221** [comment]
sphinx_book_theme suits better than pydata theme, no need to change project structure because the chapters are placed in the left sidebar. not at the top for pydata theme:

![image](https://github.com/certik/theoretical-physics/assets/102862066/7907009b-dbfc-4b2b-83c6-e79f60d379ad)

Here lists more themes:

https://sphinx-themes.org/
